### PR TITLE
build: switch to github workflows

### PR DESCRIPTION
**Pull request by itajaja**
adapted from https://github.com/python-attrs/attrs/blob/master/.github/workflows/main.yml

### CODE REVIEW

**Comment by sloria** [comment]
Can we easily get automatic releases on tagged commits?

**Comment by itajaja** [comment]
> Can we easily get automatic releases on tagged commits?

getting the test/coverage part right now. There is probably a gh action for that

File: .github/workflows/main.yml
Comment (sloria) [inline .github/workflows/main.yml:96]:
```suggestion
        run: "python -c 'import flask_resty; print(flask_resty.__version__)'"
```

File: tox.ini
Comment (sloria) [inline tox.ini:4]: question: do we not need the `[gh-actions]` section, per the docs? https://github.com/ymyzk/tox-gh-actions#tox-gh-actions-configuration. The attrs config also includes it: https://github.com/python-attrs/attrs/blob/72f94a3b03d5c2f8211f499c1c3fc72413d3fd4e/tox.ini#L11

File: .github/workflows/main.yml
Comment (sloria) [inline .github/workflows/main.yml:39]: suggestion (non-blocking): it'd be ideal if we could just run `lint` and `docs` on the latest Python version rather than all of them. No biggie though.

**Review by sloria — COMMENTED** [review]
Looking good! Just some drive-bys

File: .github/workflows/main.yml
Comment (itajaja) [reply .github/workflows/main.yml:96]: lol :swea

File: .github/workflows/main.yml
Comment (itajaja) [reply .github/workflows/main.yml:39]: yeah. inside a job there is no way to "escape" the matrix, so it would have to be done in a separate job

File: tox.ini
Comment (itajaja) [reply tox.ini:4]: yeah. so, I don't really know tox much, but as far as I can understand, doesn't using something like github actions defy the purpose of tox? gh action already has the python matrix, so seems like they fulfill similar tasks...

File: .github/workflows/main.yml
Comment (itajaja) [reply .github/workflows/main.yml:39]: ... or I could use `if` to just make it run on latest python

File: tox.ini
Comment (sloria) [reply tox.ini:4]: tox is also useful for local development. I use it for `tox -e lint` as well as reproducing CI failures. Removing Python version mgmt outside of tox does lessen the benefit, though.

File: .github/workflows/main.yml
Comment (sloria) [inline .github/workflows/main.yml:83]: _Probably_ don't need to test against mac and windows

File: .github/workflows/main.yml
Comment (sloria) [reply .github/workflows/main.yml:39]: Your call. It would be nice to separate each of the tox envs to take advantage of the parallel jobs...iirc GH allows quite a few of them (10, maybe?).

**Review by sloria — APPROVED** [review]
Approving to unblock. I'm sure there's marginal improvements to make here, but this seems to be working as is and is obviously better than the status quo.

**Comment by itajaja** [comment]
> Approving to unblock. I'm sure there's marginal improvements to make here, but this seems to be working as is and is obviously better than the status quo.

I'll figure out the tox thing first :)

File: tox.ini
Comment (c0state) [reply tox.ini:4]: Just curious--do you use `tox-pyenv` or something similar? ie: how do you have multiple clean Python installs locally to test locally that's close to CI? (or is that the wrong strategy)

**Comment by sloria** [comment]
@itajaja Are you still working on this? Would be good to get CI working 😬

**Comment by itajaja** [comment]
seems to work :) now I understand how that tox gh-actions integration is supposed to work :)

**Review by sloria — APPROVED** [review]

**Comment by sloria** [comment]
thought: we should test 3.9 too, but let's get this merged first

**Pull request merged**
